kombu/transport/mongodb.py

```diff
@@ -243,13 +243,15 @@ def _parse_uri(self, scheme='mongodb://'):
         client = self.connection.client
         hostname = client.hostname
 
-        if not hostname.startswith(scheme):
+        if '://' not in hostname:
             hostname = scheme + hostname
+        urischeme, rest = hostname.split('://', 1)
```

Comment:
`scheme` and `urischeme` seem pretty confusing. As it's not really used maybe name it `_` or just use `scheme` itself as it's the scheme?